src/vsr/replica.zig

```diff
@@ -646,9 +650,21 @@ pub fn ReplicaType(
             self.opened = false;
             self.journal.recover(journal_recover_callback);
             while (!self.opened) self.superblock.storage.tick();
+
+            // A replication quorum of replicas have committed atop op_prepare_max, op_checkpoint
+            // is guaranteed to be durable on a commit quorum of replicas.
+            if (self.op_checkpoint() == 0 or self.solo() or
+                self.commit_max > self.op_prepare_max())
+            {
+                self.checkpoint_quorum = true;
```

Comment:
It is confusing that we set `self.checkpoint_quorum` based on `op_prepare_max`, but that itself uses `self.checkpoint_quorum` inside! Perhpaps better use `prepare_max_for_checkpoint` directly here.